diff --git builds/unix/ftsystem.c builds/unix/ftsystem.c
index 3a740fd..40fa8d0 100644
--- builds/unix/ftsystem.c
+++ builds/unix/ftsystem.c
@@ -69,6 +69,9 @@
 #include <string.h>
 #include <errno.h>
 
+#ifdef VXWORKS
+#include <ioLib.h>
+#endif
 
   /*************************************************************************/
   /*                                                                       */
@@ -238,7 +241,7 @@
       return FT_Err_Invalid_Stream_Handle;
 
     /* open the file */
-    file = open( filepathname, O_RDONLY );
+    file = open( filepathname, O_RDONLY, 0);
     if ( file < 0 )
     {
       FT_ERROR(( "FT_Stream_Open:" ));
@@ -317,7 +320,11 @@
 
 
         read_count = read( file,
+#ifndef VXWORKS
                            stream->base + total_read_count,
+#else
+                           (char *) stream->base + total_read_count,
+#endif
                            stream->size - total_read_count );
 
         if ( read_count <= 0 )
